<a href="https://colab.research.google.com/github/Haritha0105/Statistical-Learning-e21172/blob/main/Assigment_05.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Foundations of Statistical Inference & Hypothesis Testing**

## **Part A: Theoretical Fundamentals (Maximum Likelihood & Decision Space)**





### Question 1: Bias of Raw MLE Covariance Matrix & Bessel's Correction

#### 1. Formal Derivation of the Bias in $\hat{\Sigma}_{MLE}$
The raw Maximum Likelihood Estimator (MLE) for the covariance matrix is defined as:
$$\hat{\Sigma}_{MLE} = \frac{1}{n} \sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T$$

To find its expectation, we can express the deviation of each sample from the sample mean $\hat{\mu}_n$ by introducing the true, unobservable population mean $\mu$:
$$X_i - \hat{\mu}_n = (X_i - \mu) - (\hat{\mu}_n - \mu)$$

Expanding the summation of outer products using this identity gives:
$$\sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T = \sum_{i=1}^{n} \left[ (X_i - \mu) - (\hat{\mu}_n - \mu) \right] \left[ (X_i - \mu) - (\hat{\mu}_n - \mu) \right]^T$$

$$= \sum_{i=1}^{n} \left( (X_i - \mu)(X_i - \mu)^T - (X_i - \mu)(\hat{\mu}_n - \mu)^T - (\hat{\mu}_n - \mu)(X_i - \mu)^T + (\hat{\mu}_n - \mu)(\hat{\mu}_n - \mu)^T \right)$$

Noting that $\sum_{i=1}^{n} (X_i - \mu) = n(\hat{\mu}_n - \mu)$, the middle cross-product terms simplify as follows:
$$\sum_{i=1}^{n} (X_i - \mu)(\hat{\mu}_n - \mu)^T = n(\hat{\mu}_n - \mu)(\hat{\mu}_n - \mu)^T$$

Substituting this back into the expansion yields:
$$\sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T = \sum_{i=1}^{n} (X_i - \mu)(X_i - \mu)^T - n(\hat{\mu}_n - \mu)(\hat{\mu}_n - \mu)^T$$

Now, we compute the mathematical expectation ($E[\cdot]$) on both sides:
$$E\left[ \sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T \right] = \sum_{i=1}^{n} E[(X_i - \mu)(X_i - \mu)^T] - n E[(\hat{\mu}_n - \mu)(\hat{\mu}_n - \mu)^T]$$

Since $X_i$ are identically and independently distributed (i.i.d.) random variables, we know that $E[(X_i - \mu)(X_i - \mu)^T] = \Sigma$. Additionally, the covariance matrix of the sample mean is given by $E[(\hat{\mu}_n - \mu)(\hat{\mu}_n - \mu)^T] = \frac{1}{n}\Sigma$.

Plugging these values into the expectation equation:
$$E\left[ \sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T \right] = n\Sigma - n\left(\frac{1}{n}\Sigma\right) = n\Sigma - \Sigma = (n-1)\Sigma$$

Thus, taking the expectation of the full $\hat{\Sigma}_{MLE}$ expression reveals its inherent downward bias in finite samples:
$$E[\hat{\Sigma}_{MLE}] = E\left[ \frac{1}{n} \sum_{i=1}^{n} (X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T \right] = \frac{1}{n} (n-1)\Sigma = \frac{n-1}{n}\Sigma$$

This completes the proof that $E[\hat{\Sigma}_{MLE}] = \frac{n-1}{n}\Sigma$.

#### 2. Constructing the Unbiased Estimator $S$ via Bessel's Correction
To eliminate the inherent sample bias, we multiply the raw MLE covariance matrix by Bessel's correction factor, which is $\frac{n}{n-1}$. This yields the unique, unbiased empirical sample estimator $S$:
$$S = \frac{n}{n-1}\hat{\Sigma}_{MLE} = \frac{n}{n-1} \left( \frac{1}{n}\sum_{i=1}^{n}(X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T \right) = \frac{1}{n-1}\sum_{i=1}^{n}(X_i - \hat{\mu}_n)(X_i - \hat{\mu}_n)^T$$

Verifying its unbiased property:
$$E[S] = E\left[ \frac{n}{n-1}\hat{\Sigma}_{MLE} \right] = \frac{n}{n-1} E[\hat{\Sigma}_{MLE}]$$

Substituting our derived expectation $E[\hat{\Sigma}_{MLE}] = \frac{n-1}{n}\Sigma$:
$$E[S] = \frac{n}{n-1} \left( \frac{n-1}{n}\Sigma \right) = \Sigma$$

This demonstrates mathematically that applying Bessel's correction successfully yields an unbiased covariance matrix estimator $S$.

---

### Question 2: Diagnostic Errors & Threshold Consequences

#### 1. Definitions within Continuous Structural Health Monitoring (SHM)
* **Type I Error ($\alpha$):** Occurs when the framework falsely rejects the null hypothesis $H_0$ when it is actually true. In an SHM context, this is a **False Alarm**, meaning the diagnostic layer concludes that a healthy asset is damaged or undergoing structural failure, triggering unneeded emergency shutdowns and inspections.
* **Type II Error ($\beta$):** Occurs when the framework fails to reject the null hypothesis $H_0$ when it is false. In an SHM context, this is a **Missed Detection**, meaning the diagnostic system concludes that the asset is completely healthy when it actually contains active structural degradation, risking catastrophic in-service failure.

#### 2. Mathematical Consequence of an Ultra-Conservative Threshold ($\alpha = 0.0001$)
When an engineer sets an exceptionally low significance threshold ($\alpha = 0.0001$) to prevent costly false-alarm plant shutdowns, it drastically reduces the size of the critical rejection region.

Because the probability of a Type I error ($\alpha$) and a Type II error ($\beta$) are inversely proportional for a fixed sample size, forcing $\alpha \to 0$ causes **the probability of a Type II error ($\beta$) to rise significantly**. Since the statistical power of a diagnostic layer is defined as:
$$\text{Power} = 1 - \beta$$

This decision results in a severe **collapse of the statistical power** of your monitoring layer. The system becomes near-blind to real, early-stage structural fractures or progressive shifts away from the operational baseline because the threshold requires overwhelming evidence to flag any anomaly.

#### 3. Modification to the Geometric Volume of the Confidence Ellipsoid
The geometric volume of the "healthy operation" confidence ellipsoid **expands significantly**.

To guarantee that a healthy status quo observation vector has only a $0.01\%$ chance ($\alpha = 0.0001$) of falling outside the baseline boundaries, the Mahalanobis distance threshold must be pushed far outward into the tails of the distribution. This inflates the multi-dimensional confidence envelope to encapsulate nearly all extreme noise fluctuations, reducing the diagnostic spatial sensitivity to legitimate structural failures.

## **Part B: Theoretical Extension (Asymptotic Distributions & Slutsky's Theorem)**



### Question 1: Mathematical Statement of Slutsky's Theorem

Let $X_n$ and $Y_n$ be sequences of random vectors or matrices defined on a common probability space.

If $X_n$ converges in distribution to a random variable $X$ as $n \to \infty$, denoted as:
$$X_n \xrightarrow{d} X$$

And $Y_n$ converges in probability to a non-random constant matrix $c$ as $n \to \infty$, denoted as:
$$Y_n \xrightarrow{p} c$$

Then, the product sequence $Y_n X_n$ converges in distribution to the transformed random vector $cX$:
$$Y_n X_n \xrightarrow{d} cX$$

---

### Question 2: Substitution Proof using the Central Limit Theorem (CLT)

#### 1. Setup of the Initial Limit
By applying the Lindeberg-Lévy Multivariate Central Limit Theorem (CLT) to our sequence of independent and identically distributed (i.i.d.) random vectors $X_i$ with population mean $\mu$ and true population covariance matrix $\Sigma$, we establish that:
$$\sqrt{n}(\hat{\mu}_n - \mu) \xrightarrow{d} N(0, \Sigma)$$

To standardize this multivariate distribution to a standard normal form $N(0, I_m)$, we premultiply the sequence by the inverse matrix square root of the population covariance matrix, $\Sigma^{-1/2}$:
$$\Sigma^{-1/2} \sqrt{n}(\hat{\mu}_n - \mu) \xrightarrow{d} N(0, \Sigma^{-1/2}\Sigma\Sigma^{-1/2}) = N(0, I_m)$$

#### 2. Application of the Weak Law of Large Numbers (WLLN)
In practical monitoring environments, the true population covariance matrix $\Sigma$ is a latent parameter and remains unobservable. Instead, we compute the empirical sample covariance matrix $S$.

By the Weak Law of Large Numbers (WLLN), the sample covariance matrix $S$ is a consistent estimator of the true covariance matrix $\Sigma$:
$$S \xrightarrow{p} \Sigma \quad \text{as } n \to \infty$$

Since matrix inversion and taking the matrix square root are continuous operations over the space of non-singular, positive-definite matrices, we apply the Continuous Mapping Theorem (CMT) to deduce that:
$$S^{-1/2} \xrightarrow{p} \Sigma^{-1/2} \quad \text{as } n \to \infty$$

#### 3. Combining Sequences via Slutsky's Theorem
We now define our two target sequences to match the framework of Slutsky's Theorem:
* Let the sequence of random vectors be $A_n = \sqrt{n}(\hat{\mu}_n - \mu)$, where $A_n \xrightarrow{d} N(0, \Sigma)$.
* Let the sequence of random matrices be $B_n = S^{-1/2}$, where $B_n \xrightarrow{p} \Sigma^{-1/2}$.

Applying Slutsky's Theorem directly to the matrix-vector product $B_n A_n$:
$$S^{-1/2} \sqrt{n}(\hat{\mu}_n - \mu) \xrightarrow{d} \Sigma^{-1/2} N(0, \Sigma)$$

Since linear transformations of a multivariate normal vector yield another multivariate normal vector, the asymptotic distribution simplifies exactly to:
$$S^{-1/2} \sqrt{n}(\hat{\mu}_n - \mu) \xrightarrow{d} N(0, \Sigma^{-1/2} \Sigma \Sigma^{-1/2}) = N(0, I_m)$$

#### 4. Practical Engineering Justification
This asymptotic convergence justifies that for a sufficiently large number of structural observations ($n \to \infty$), the unobservable population covariance matrix $\Sigma$ can be substituted directly with the observable empirical matrix $S$ without losing statistical validity.

Consequently, we can isolate the sample mean vector $\hat{\mu}_n$ to frame our continuous diagnostic parametric testing model as:
$$\hat{\mu}_n \sim N\left(\mu, \frac{1}{n}S\right)$$

This completes the formal theoretical extension.

## **Part C: Numerical Verification**

In [5]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# --- DO NOT MODIFY: Synthetic Strain Sensor Data Generator ---
np.random.seed(42)
n_samples = 5000
n_features = 3

base_data = np.random.multivariate_normal(
    mean=[0.5, -0.2, 1.1],
    cov=[[0.09, 0.02, 0.01],
         [0.02, 0.06, 0.03],
         [0.01, 0.03, 0.05]],
    size=n_samples
)
# Injecting a subtle, hidden structural drift in the final 1,000 snapshots
base_data[4000:, 0] += 0.015
base_data[4000:, 2] -= 0.010

df_strain = pd.DataFrame(base_data, columns=['Strain_Chl', 'Strain_Ch2', 'Strain_Ch3'])
# ----------------------------------------------------------------

def verify_first_moment_homogeneity(df: pd.DataFrame, g_chunks: int = 5) -> dict:
    """
    Rigorously tests the stationarity of the multivariate first moment (mean vector)
    by partitioning the time-series matrix into g non-overlapping chronological chunks.
    """
    n = len(df)
    m = df.shape[1]
    n_j = n // g_chunks  # Sample size per equal-sized chunk

    # Compute the global coordinate mean vector
    global_mean = df.mean().values

    # Initialize Within-Group (W) and Between-Group (B) matrices
    W = np.zeros((m, m))
    B = np.zeros((m, m))

    # Extract chronological segments and accumulate matrices
    for g in range(g_chunks):
        chunk = df.iloc[g * n_j : (g + 1) * n_j]
        chunk_mean = chunk.mean().values

        # Within-Chunk Sum of Squares
        chunk_centered = chunk.values - chunk_mean
        W += chunk_centered.T @ chunk_centered

        # Between-Chunk Sum of Squares
        mean_diff = (chunk_mean - global_mean).reshape(-1, 1)
        B += n_j * (mean_diff @ mean_diff.T)

    # Compute Wilks' Lambda
    wilks_lambda = np.linalg.det(W) / np.linalg.det(B + W)

    # Bartlett's Chi-Square asymptotic transformation
    multiplier = -(n - 1 - (m + g_chunks) / 2)
    chi_calc = multiplier * np.log(wilks_lambda)

    # Degrees of Freedom
    dof = m * (g_chunks - 1)

    # Asymptotic p-value
    p_value = 1 - stats.chi2.cdf(chi_calc, dof)

    return {
        "Wilks_Lambda": wilks_lambda,
        "Bartlett_Chi2": chi_calc,
        "Degrees_of_Freedom": dof,
        "p_value": p_value
    }

# Execute verification pipeline
metrics = verify_first_moment_homogeneity(df_strain, g_chunks=5)
print("--- Execution Outputs ---")
for key, value in metrics.items():
    print(f"{key}: {value}")

--- Execution Outputs ---
Wilks_Lambda: 0.9904519796929591
Bartlett_Chi2: 47.9215049961488
Degrees_of_Freedom: 12
p_value: 3.2255259508895406e-06


### 2. Tabulated Mathematical Outputs

Running the MANOVA homogeneity engine on the `df_strain` matrix yields the following exact quantitative breakdown:

| Metric | Mathematical Notation | Computed Empirical Value |
| :--- | :---: | :--- |
| **Wilks' Lambda** | $\Lambda$ | `0.9904519796929591` |
| **Bartlett $\chi^2$ Statistic** | $\chi^2_{calc}$ | `47.9215049961488` |
| **Degrees of Freedom** | $\nu$ | `12` |
| **Asymptotic $p$-value** | $p$ | `3.2255259508895406e-06` |

---

### 3. Structural Health Verification & Hypothesis Conclusion

#### 1. Hypothesis Formulation
To evaluate the stationarity of the operational baseline center across the chronological lifecycle, we establish the following statistical criteria:
* $H_0: \mu_1 = \mu_2 = \mu_3 = \mu_4 = \mu_5$ (The multi-channel mean strain vector remains perfectly invariant across all sequential blocks, establishing structural homogeneity).
* $H_1: \mu_j \neq \mu_k$ for at least one pair of blocks $j \neq k$ (The mean vector exhibits a statistically significant temporal shift, indicating an active structural baseline drift).

#### 2. Decision Logic and Boundary Boundary Rule
We define the structural risk threshold at a standard significance level of $\alpha = 0.05$.

The theoretical degrees of freedom for the critical chi-square boundary are calculated as:
$$\nu = m(g - 1) = 3 \times (5 - 1) = 12$$

Using the theoretical limiting distribution, the upper-tail critical rejection boundary is defined as:
$$\chi^2_{crit}(0.05, 12) \approx 21.026$$

The core evaluation framework maps as:
$$\text{Reject } H_0 \iff \chi^2_{calc} > \chi^2_{crit} \quad \text{or} \quad p \le \alpha$$

#### 3. Diagnostic Inference
Comparing the empirical results derived by our processing engine against the theoretical decision bounds:
$$\chi^2_{calc} = 47.922 > 21.026$$
$$p = 3.226 \times 10^{-6} \ll \alpha = 0.05$$

Because the sample test statistic sits exceptionally deep within the critical right-tail rejection zone, **we reject the null hypothesis $H_0$ with near absolute statistical certainty**.

#### 4. Physical Engineering Implications
The rejection of $H_0$ delivers definitive empirical proof that the multi-sensor network is **non-homogeneous** over time. Even though the injected anomaly was intentionally designed to be minor and latent ($+0.015$ on Channel 1 and $-0.010$ on Channel 3 during the final 1,000 snapshots), the MANOVA framework successfully distinguished this shift from common Gaussian measurement noise.

From an asset-integrity perspective, **this baseline cannot be certified as stationary**. The diagnostic layer confirms that a statistically significant physical drift or structural alteration has occurred within the operational baseline, signaling the engineering team to log a baseline variation event.

# **Geometric Subspace Optimization via Principal Component Analysis (PCA)**

## **Part A: Theoretical Fundamentals (Coordinate Projections & Orthogonality)**



### Question 1: Diagonal Matrix Simplification Proof

#### 1. Formal Proof that $E[Z_i Z_i^T] = \Lambda$
[cite_start]We are given a zero-mean, centered baseline feature vector $\tilde{X}_i = (X_i - \mu) \in \mathbb{R}^m$[cite: 84]. The symmetric, positive-semidefinite population covariance matrix is defined as:
$$\Sigma = E[\tilde{X}_i \tilde{X}_i^T]$$

Applying the spectral decomposition theorem to $\Sigma$, we express the covariance matrix in terms of its orthogonal eigenvector matrix $P$ and diagonal eigenvalue matrix $\Lambda$:
$$\Sigma = P \Lambda P^T$$

The orthogonal projection mapping defines the transformed principal component coordinate vector $Z_i$ as:
$$Z_i = P^T \tilde{X}_i$$

To find the theoretical covariance matrix of the transformed coordinate vector, we expand the mathematical expectation of its outer product:
$$E[Z_i Z_i^T] = E\left[ (P^T \tilde{X}_i)(P^T \tilde{X}_i)^T \right]$$

Utilizing the linear algebra transpose property $(AB)^T = B^T A^T$, the second term inside the expectation expands to $(P^T \tilde{X}_i)^T = \tilde{X}_i^T (P^T)^T = \tilde{X}_i^T P$. Substituting this back yields:
$$E[Z_i Z_i^T] = E\left[ P^T \tilde{X}_i \tilde{X}_i^T P \right]$$

Since the matrix of eigenvectors $P$ consists of deterministic constants, the linear expectation operator $E[\cdot]$ distributes strictly across the random vectors:
$$E[Z_i Z_i^T] = P^T E[\tilde{X}_i \tilde{X}_i^T] P$$

Substituting the defined population covariance matrix $\Sigma = E[\tilde{X}_i \tilde{X}_i^T]$:
$$E[Z_i Z_i^T] = P^T \Sigma P$$

Now, we substitute the spectral decomposition identity $\Sigma = P \Lambda P^T$ into the expression:
$$E[Z_i Z_i^T] = P^T (P \Lambda P^T) P = (P^T P) \Lambda (P^T P)$$

Because $P$ is an orthogonal matrix, its columns are orthonormal vectors, establishing the property $P^T P = P P^T = I_m$, where $I_m$ is the identity matrix. Simplifying the expression gives:
$$E[Z_i Z_i^T] = I_m \Lambda I_m = \Lambda$$

[cite_start]This completes the step-by-step mathematical proof that $E[Z_i Z_i^T]$ simplifies exactly to the strictly diagonal eigenvalue matrix $\Lambda$[cite: 86].

#### 2. Geometric and Statistical Interpretation
* [cite_start]**Statistical Meaning:** The strictly diagonal result proves that the cross-covariances between distinct coordinates $Z_{i,j}$ and $Z_{i,k}$ (for $j \neq k$) are precisely zero[cite: 86, 87]. [cite_start]This implies that the original highly collinear multi-sensor variables have been completely de-correlated into mutually orthogonal dimensions[cite: 224].
* [cite_start]**Geometric Meaning:** The orthogonal transformation $P^T$ acts as a pure rigid rotation of the high-dimensional coordinate framework[cite: 84]. It aligns the new coordinate axes ($Z$) perfectly with the directions of maximum variance in the original multi-dimensional dataset, meaning the data cloud's main geometric axes now line up directly with our standardized basis vectors.

---

### Question 2: Invariance of Total Variance under Orthogonal Projections

[cite_start]To formally prove that the total system variance is preserved during the mapping from raw space to the de-correlated principal space, we exploit the properties of the matrix trace operator ($tr$)[cite: 88, 89].

[cite_start]The total variance in the raw coordinate space is represented by the trace of the population covariance matrix, $tr(\Sigma)$[cite: 88, 89]. Substituting the spectral decomposition $\Sigma = P \Lambda P^T$:
$$tr(\Sigma) = tr(P \Lambda P^T)$$

The trace operator exhibits an algebraic property known as **invariance under cyclic permutations**, which states that for any matrices $A$ and $B$ where the dimensions allow multiplication:
$$tr(ABC) = tr(CAB) = tr(BCA)$$

Letting $A = P$, $B = \Lambda$, and $C = P^T$, we cyclically permute the matrix string by moving $P^T$ from the end to the front of the trace product:
$$tr(P \Lambda P^T) = tr(P^T P \Lambda)$$

Because $P$ is an orthogonal projection matrix, the inner product $P^T P$ collapses directly into the identity matrix $I_m$:
$$tr(P^T P \Lambda) = tr(I_m \Lambda) = tr(\Lambda)$$

Since $\Lambda$ is a strictly diagonal matrix containing the individual eigenvalues $\lambda_j$, its trace is simply the summation of its diagonal entries:
$$tr(\Lambda) = \sum_{j=1}^{m} \lambda_j$$

Therefore, we have established the absolute conservation of total variance:
$$tr(\Sigma) = tr(\Lambda) = \sum_{j=1}^{m} \lambda_j$$

[cite_start]This completes the proof that total system variance is perfectly preserved during the geometric rotation into principal component space[cite: 89, 90].

---

### Question 3: Variance Partitioning Formulations & Subspace Norm Decomposition

#### 1. Rigorous Formulations for Cumulative Variance Ratios
[cite_start]Given a partitioned empirical system with an explained principal subspace $\hat{P}_k \in \mathbb{R}^{m \times k}$ and an unexplained residual subspace $\hat{P}_{m-k} \in \mathbb{R}^{m \times (m-k)}$[cite: 91]:
* **Cumulative Explained Variance Ratio ($\Phi(k)$):** Measures the proportion of global system energy successfully captured by retaining the first $k$ principal dimensions:
$$\Phi(k) = \frac{\sum_{j=1}^{k} \lambda_j}{tr(\Sigma)} = \frac{\sum_{j=1}^{k} \lambda_j}{\sum_{j=1}^{m} \lambda_j}$$
* **Residual Unexplained Variance Ratio ($\Psi(k)$):** Measures the relative scale of structural information discarded or filtered out when truncating the system subspace to size $k$:
$$\Psi(k) = 1 - \Phi(k) = \frac{\sum_{j=k+1}^{m} \lambda_j}{\sum_{j=1}^{m} \lambda_j}$$

#### 2. Vector Partitioning Setup
[cite_start]When an asset monitoring model is truncated to $k$ components ($k < m$), the centered sample observation vector is split into a reconstructed vector $\hat{x}_i$ and a residual error vector $e_i$[cite: 93, 94, 96]:
$$(x_i - \hat{\mu}_n) = \underbrace{\hat{P}_k z_{i,k}}_{\hat{x}_i} + \underbrace{\hat{P}_{m-k} z_{i,m-k}}_{e_i}$$

[cite_start]Where $\hat{x}_i = \hat{P}_k z_{i,k}$ is the projection onto the clean principal subspace, and $e_i = \hat{P}_{m-k} z_{i,m-k}$ represents the projection onto the orthogonal residual space[cite: 95, 96].

#### 3. Subspace Decomposition Proof via Pythagorean Identity
[cite_start]To evaluate the total squared Euclidean norm of the centered observation vector, we compute its inner product[cite: 97, 98]:
$$||x_i - \hat{\mu}_n||^2 = (x_i - \hat{\mu}_n)^T (x_i - \hat{\mu}_n)$$

Substituting the partitioned vector expressions $\hat{x}_i + e_i$:
$$||x_i - \hat{\mu}_n||^2 = (\hat{x}_i + e_i)^T (\hat{x}_i + e_i)$$
$$= \hat{x}_i^T \hat{x}_i + \hat{x}_i^T e_i + e_i^T \hat{x}_i + e_i^T e_i$$
$$= ||\hat{x}_i||^2 + 2\hat{x}_i^T e_i + ||e_i||^2$$

We examine the cross-product term $\hat{x}_i^T e_i$ by substituting their projection definitions:
$$\hat{x}_i^T e_i = (\hat{P}_k z_{i,k})^T (\hat{P}_{m-k} z_{i,m-k}) = z_{i,k}^T \hat{P}_k^T \hat{P}_{m-k} z_{i,m-k}$$

By definition, the full spectral decomposition matrix $\hat{P} = [\hat{P}_k \mid \hat{P}_{m-k}]$ is orthogonal. Therefore, any column in the explained subspace $\hat{P}_k$ is strictly orthogonal to any column in the residual subspace $\hat{P}_{m-k}$, forcing their matrix inner product to zero:
$$\hat{P}_k^T \hat{P}_{m-k} = \mathbf{0}_{k \times (m-k)}$$

Because the subspaces are perfectly orthogonal, the cross-product term vanishes completely:
$$\hat{x}_i^T e_i = z_{i,k}^T (\mathbf{0}) z_{i,m-k} = 0$$

Now, looking at the individual squared norm terms:
$$||\hat{x}_i||^2 = (\hat{P}_k z_{i,k})^T (\hat{P}_k z_{i,k}) = z_{i,k}^T (\hat{P}_k^T \hat{P}_k) z_{i,k}$$
Since the columns within $\hat{P}_k$ are orthonormal, $\hat{P}_k^T \hat{P}_k = I_k$, leaving:
$||\hat{x}_i||^2 = z_{i,k}^T I_k z_{i,k} = ||z_{i,k}||^2$

Similarly, for the error vector:
$$||e_i||^2 = (\hat{P}_{m-k} z_{i,m-k})^T (\hat{P}_{m-k} z_{i,m-k}) = z_{i,m-k}^T (\hat{P}_{m-k}^T \hat{P}_{m-k}) z_{i,m-k} = ||z_{i,m-k}||^2$$

Substituting these back into the Pythagorean expansion proves that total energy decomposes cleanly into independent subspace energies:
$$||x_i - \hat{\mu}_n||^2 = ||z_{i,k}||^2 + ||z_{i,m-k}||^2$$

[cite_start]This completes the formal proof[cite: 98, 99].

---

### Question 4: Diagnostic Anomaly Mapping ($T^2$ vs. $Q$ Statistic)

#### 1. Core Contrast in Diagnostic Functions
* [cite_start]**Hotelling's $T^2$ Statistic:** Measures the distance from the baseline center *within* the major retained principal subspace[cite: 79, 100]. [cite_start]It evaluates whether the operational state has shifted excessively along established, nominal structural pathways[cite: 102].
* [cite_start]**Residual $Q$ Statistic (SPE):** Measures the orthogonal distance *out of* the principal subspace[cite: 79, 100]. [cite_start]It tracks the energy that can no longer be explained by the baseline PCA model, effectively monitoring shifts in the underlying correlation structure itself[cite: 100].

#### 2. Anomaly Event Mapping Matrix

| Physical Event Type | Target Sensitive Statistic | Engineering & Mathematical Justification |
| :--- | :---: | :--- |
| **Environmental Load Anomaly** *(e.g., extreme wind gusts)* | [cite_start]**Hotelling's $T^2$** [cite: 101, 102] | High operational wind loads force the asset to flex significantly, but it continues to follow its normal physical design paths. [cite_start]The cross-sensor correlations remain unchanged, but the data points move out into the extreme tails of the retained subspace, making the $T^2$ distance metric highly sensitive to the event[cite: 101, 102]. |
| **Internal Structural Fracture** *(e.g., progressive fatigue crack)* | [cite_start]**Residual $Q$ Statistic (SPE)** [cite: 102] | [cite_start]A physical fracture alters the structural load pathways of the asset[cite: 102]. [cite_start]This violates the baseline sensor cross-correlation matrix ($\Sigma$)[cite: 84]. [cite_start]Because the data points now move in an entirely new geometric direction that the model has never seen, the projection fails to line up with $\hat{P}_k$, creating a sharp spike in the residual error vector ($e_i$) tracked by $Q$[cite: 95, 96, 100]. |

## **Part B: Parametric Factor Partition Engine & UI Architecture**



### 1. Architectural System Overview

[cite_start]The primary objective is to implement a production-grade, end-to-end data processing and diagnostic visualization pipeline that instantiates a parametric Factor Analysis (FA) engine with Varimax orthogonal rotation[cite: 141].

[cite_start]The architecture partitions empirical observation metrics into a shared structural subspace (communalities) and a distinct diagonal noise floor (uniquenesses), mapping the downstream telemetry onto a highly structured, asymmetric grid interface[cite: 142, 144].

---

### 2. Mathematical Pre-processing & Guardrail Layer

[cite_start]Before executing the decomposition matrix calculations, incoming multivariate datasets must pass through three decoupled validation and standardization layers[cite: 144]:

#### 1. Z-Score Mapping & Scale Normalization
[cite_start]Accepting an arbitrary data matrix $X \in \mathbb{R}^{n \times m}$, the processing layer isolates the numeric observation columns and standardizes them into standard normal space $Z \in \mathbb{R}^{n \times m}$[cite: 146, 147]. [cite_start]This uses the unbiased sample standard deviation with degrees of freedom correction ($ddof=1$)[cite: 147]:
$$Z_{i,j} = \frac{X_{i,j} - \hat{\mu}_j}{\sigma_j}$$
$$\text{where} \quad \hat{\mu}_j = \frac{1}{n}\sum_{i=1}^{n}X_{i,j} \quad \text{and} \quad \sigma_j = \sqrt{\frac{1}{n-1}\sum_{i=1}^{n}(X_{i,j} - \hat{\mu}_j)^2}$$

#### 2. Numerical Guardrail Execution
[cite_start]To guarantee runtime robustness and eliminate mathematical exceptions caused by division-by-zero during normalization, an isolated guardrail checks for zero-variance channels ($\sigma_j = 0$)[cite: 148, 151]. [cite_start]If detected, the scale divisor is capped to a minimum boundary threshold[cite: 151]:
$$\sigma_j \leftarrow \max\left(\sigma_j, 10^{-15}\right)$$

#### 3. Dimensionality Constraint Layer
[cite_start]Let $m$ represent the isolated sensor channels, and $k$ represent the target count of latent physical factors[cite: 152]. [cite_start]To prevent underdetermined matrix systems, two strict architectural validation check rules are enforced[cite: 153]:
* [cite_start]**Snapshot Exception Rule:** Reject execution and throw an error if the sample snapshot size ($n$) is less than or equal to the channel count ($m$)[cite: 154]:
$$n \le m \implies \text{HALT\_PROCESSING}$$
* [cite_start]**Latent Space Dimension Rule:** Throw a descriptive exception if the requested latent space dimension ($k$) is not strictly less than the available feature space dimension ($m$)[cite: 155]:
$$k \ge m \implies \text{RAISE\_CONSTRAINT\_ERROR}$$

---

### 3. Factor Analysis Information Decomposition Layer

[cite_start]The framework fits the standardized snapshot matrix $Z$ using a Maximum Likelihood or Singular Value Decomposition (SVD) algorithm combined with an orthogonal Varimax rotation matrix $T$[cite: 156]. [cite_start]From the fitted factor loadings $\Lambda_{rotated}$, the engine isolates the following metrics[cite: 157]:

#### 1. Communality Vector ($h^2 \in \mathbb{R}^m$)
[cite_start]The total shared variance captured by the $k$ latent dimensions across each channel $j$[cite: 158]:
$$h_j^2 = \sum_{r=1}^{k} \lambda_{j,r}^2$$

#### 2. Uniqueness Vector ($\varphi^2 \in \mathbb{R}^m$)
[cite_start]The remaining isolated localized sensor noise floor that operates outside the shared physical process space[cite: 159]:
$$\varphi_j^2 = 1 - h_j^2$$

---

### 4. Thomson's Latent Score Projection Matrix

#### 1. Mathematical State Reconstruction
Because common factors are hidden parameters, they cannot be extracted via simple geometric projections. [cite_start]Instead, we reconstruct the hidden factor states $F \in \mathbb{R}^{n \times k}$ using Thomson's Minimum Mean Squared Error (MMSE) regression methodology[cite: 135, 163]:
$$F = Z R^{-1} \Lambda_{rotated}$$
$$\text{where} \quad R = \frac{1}{n-1}Z^T Z \in \mathbb{R}^{m \times m} \quad \text{is the empirical correlation matrix.}$$

#### 2. Performance Diagnostic Telemetry
[cite_start]To verify that the variance of the reconstructed factor scores matches the constraints of the generative model, the engine computes the empirical variance of $F$ enforcing an unbiased scale correction ($ddof=1$)[cite: 165]. [cite_start]It outputs the global metrics to the console for tracking[cite: 166]:
$$\text{Average System Communality \%} = \left(\frac{1}{m}\sum_{j=1}^{m}h_j^2\right) \times 100$$
$$\text{Average System Uniqueness \%} = \left(\frac{1}{m}\sum_{j=1}^{m}\varphi_j^2\right) \times 100$$

---

### 5. Layout Engineering & Interface Design Canvas

[cite_start]The pipeline renders a production-grade $2 \times 2$ diagnostic dashboard using Plotly Subplots, enforcing strict geometric layout parameters to optimize readability and isolate interface elements[cite: 172]:

* [cite_start]**Grid Dimensions:** `horizontal_spacing = 0.24`, `vertical_spacing = 0.28`, `width = 1250`, `height = 750`, `theme = plotly_white`[cite: 173].
* [cite_start]**Legend Alignment:** `orientation = "h"`, `x = 0.5`, `y = 1.02` (placed horizontally directly above subplots)[cite: 174].
* [cite_start]**Canvas Margin Padding:** `top = 150`, `bottom = 60`, `left = 140`, `right = 80`[cite: 175].

#### [cite_start]Interface Panel Matrix Map ($2 \times 2$) [cite: 177]

# **Subspace Diagnostics & Feature De-correlation**

## **Part A: Develop a Feature Analysis Dashboard**



### 1. Architectural System Overview

[cite_start]The primary objective of this diagnostic module is to build a comprehensive data processing and visualization pipeline inside Google Colab that evaluates two fundamental methods for dimensionality reduction using an identical 4-channel sensor array (`Sensor_1` to `Sensor_4`)[cite: 226]:
1. [cite_start]**Principal Component Analysis (PCA):** A geometric, non-parametric approach aimed at total variance maximization[cite: 229].
2. [cite_start]**Factor Analysis (FA):** A generative, parametric framework aimed at partitioning shared structural variance from unique localized sensor noise[cite: 230].

The processing engine computes the mathematical features for both paradigms from scratch and projects the results onto two highly optimized dashboard interfaces implemented via Plotly Subplots.

---

### 2. UI Layout 1: PCA Optimization Suite ($2 \times 3$ Matrix Grid)

[cite_start]The first interface structures its telemetry into a 5-panel horizontal grid layout to monitor structural energy levels and optimization criteria across different truncation boundaries $k \in \{1, 2, 3\}$[cite: 105, 232, 233].

#### PCA Layout Specifications
* **Grid Geometry:** $2 \times 3$ subplot matrix with an explicit `horizontal_spacing = 0.08` and `vertical_spacing = 0.12`.
* **Canvas Parameters:** Global width scaled to $1500$ pixels and height to $900$ pixels utilizing the `plotly_white` production theme.
* **Legend Alignment:** Placed horizontally directly above the upper row (`orientation = "h"`, `x = 0.5`, `y = 1.05`).

#### PCA Panel Allocation Map

## **Part B: Analysis**



### Question 1: The Total Variance Illusion (PCA Subplots 1 & 2 vs. FA Subplots 1 & 2)

#### 1. Physical Nature of the Data Coming from Sensor_4
[cite_start]A Uniqueness value ($\varphi^2$) exceeding $98\%$ tells us that almost all the variance ($>98\%$) observed in `Sensor_4` is completely isolated, localized noise unique to that specific instrument[cite: 271, 272]. [cite_start]It means that `Sensor_4` fails to capture or couple with the underlying, shared physical driving forces ($f_1$ and $f_2$) governing the actual mechanical structure[cite: 255, 272]. [cite_start]Physically, this indicates an extreme electrical/instrumentation fault, such as a severe calibration loss, a damaged transducer, or high-scale localized white noise contamination[cite: 260, 268].

#### 2. How the Massive Localized Noise of Sensor_4 "Tricks" the PCA Engine
[cite_start]Principal Component Analysis (PCA) operates strictly as a non-parametric, geometric variance maximizer without distinguishing between structural variance (covariance shared across sensors) and localized noise variance (uniqueness)[cite: 229, 230, 273].

[cite_start]Because `Sensor_4` possesses an exceptionally large total variance magnitude ($\sigma^2 \approx 2.0$) due to its high noise contamination [cite: 261, 268][cite_start], the global optimization framework of PCA prioritizes sweeping up this sheer energy scale to maximize total global variance[cite: 229, 273]. [cite_start]Consequently, the PCA engine is "tricked" into inflating its top eigenvalues ($\lambda_j$), artificially allocating a large percentage of explained variance to the primary principal components (e.g., PC 1 and PC 2)[cite: 235, 270, 273]. [cite_start]In reality, these components are capturing meaningless instrument noise rather than genuine multi-sensor physical modes[cite: 260, 270].

#### 3. Major Risks in Automated Plant Anomaly Detection Frameworks
[cite_start]If an engineer relies solely on a standard PCA framework, the automated plant monitoring layer inherits two severe risks[cite: 274]:
* [cite_start]**High Rate of Missed Detections (Blindness to Real Damage):** Since the top eigenvalues are dominated and inflated by the instrumentation noise of a single bad sensor, a legitimate, subtle structural failure in the asset will be masked[cite: 260, 273]. [cite_start]The system will fail to generate an anomaly flag because the true structural energy shifts are tiny compared to the massive noise scale driving the principal subspace[cite: 261, 273].
* [cite_start]**High Rate of False Alarms:** Any minor fluctuation, electrical surge, or transient pop within the faulty sensor (`Sensor_4`) will map directly as a massive distance shift in the principal subspace, triggering unneeded emergency shutdowns and costly inspections for a healthy asset[cite: 17, 261].

---

### Question 2: Decoupling Structural Loading via Rotation (FA Subplot 1 vs. PCA Eigenvectors)

#### 1. How Varimax Rotation Relaxes the Strict PCA Hierarchy to Achieve "Simple Structure"
[cite_start]Traditional PCA forces its eigenvectors to follow a rigid mathematical hierarchy where each successive component must maximize the remaining mixed global variance under strict global orthogonality constraints ($\lambda_1 > \lambda_2 > \dots$)[cite: 229, 276]. [cite_start]This results in complex, cross-coupled eigenvectors where every sensor shows non-zero loadings across all components, making physical interpretation difficult[cite: 276].

[cite_start]Factor Analysis with a Varimax rotation relaxes this strict global variance maximization constraint[cite: 230, 277]. [cite_start]Varimax is an orthogonal coordinate rotation that maximizes the **variance of the squared loadings within each column** of the factor matrix[cite: 131, 277]. [cite_start]Mathematically, it pushes the factor loadings toward the extremes: making them either very high (close to 1 or -1) or very low (close to 0)[cite: 277]. [cite_start]This drives the system toward a **"simple structure,"** where each sensor channel maps cleanly and heavily onto exactly one latent physical factor, completely isolating independent physical processes[cite: 277].

#### 2. Troubleshooting Benefits for a Plant Operator
[cite_start]From a live plant operator's standpoint, the rotated FA loading heatmap is significantly easier to troubleshoot during a structural failure than a raw PCA eigenvector matrix[cite: 278]:
* [cite_start]Under the **Rotated FA Heatmap**, the channels are cleanly decoupled [cite: 277][cite_start]: `Sensor_1` and `Sensor_2` map exclusively to Factor 1 (Primary Structural Mode), while `Sensor_3` maps exclusively to Factor 2 (Secondary Operational Mode)[cite: 256, 275, 276]. [cite_start]If an anomaly spikes specifically along Factor 1, the operator knows instantly that the physical failure is localized to the exact zone monitored by Sensors 1 and 2[cite: 258, 285].
* [cite_start]Under a **Raw PCA Matrix**, the loading weights are mathematically mixed across components[cite: 276]. [cite_start]A failure on a single physical mode will bleed across multiple principal axes simultaneously, forcing the operator to decipher complex linear combinations to figure out which physical component or sensor cluster is actually degrading[cite: 276].

---

### Question 3: Determining Subspace Truncation ($k$) using $T^2$ and $Q$ Profiles

#### 1. Exact Behavior of the Mean $Q$ Statistic Curve
* [cite_start]**From $k=1$ to $k=2$:** The Mean $Q$ Statistic (SPE) curve exhibits a steep, dramatic drop[cite: 240, 280]. [cite_start]This occurs because moving the truncation boundary to $k=2$ allows the subspace to capture the secondary independent physical process ($f_2$), which accounts for a substantial portion of the structured sensor cross-correlations[cite: 256, 259, 267].
* [cite_start]**From $k=2$ to $k=3$:** The Mean $Q$ Statistic curve flattens completely, forming a clear, sharp "elbow" or plateau[cite: 240, 280, 281]. [cite_start]Discarding or retaining a third dimension provides virtually zero additional reduction in residual error energy[cite: 240, 280].

#### 2. Why the "Flat Elbow" Identifies $k=2$ as the True Hidden Dimensionality
[cite_start]The $Q$ statistic measures the remaining residual energy that cannot be explained by the retained subspace components[cite: 100, 240]. [cite_start]A sharp drop followed by a distinct "flat elbow" indicates that the first two dimensions ($k=2$) successfully sweep up all structured, highly correlated physical cross-covariance present across the multi-sensor grid[cite: 267, 281]. [cite_start]Beyond $k=2$, the remaining residual energy consists strictly of unstructured, independent white noise[cite: 258, 261]. [cite_start]Because white noise cannot be compressed or linearly modeled, adding more dimensions does not cause further decay in the residual curve, mathematically confirming that the underlying physical process space has a true dimensionality of exactly $2$[cite: 255, 281].

#### 3. Consequence of Incorrectly Choosing a Truncation Cutoff of $k=3$
[cite_start]If an engineer incorrectly selects a truncation cutoff of $k=3$, they are forcing the system to expand its "clean" principal subspace to capture a third dimension[cite: 282]. [cite_start]Because the true physical data space is completely exhausted at $k=2$, this third component is forced to model the remaining unstructured residual energy[cite: 255, 281, 282].

[cite_start]Consequently, the framework **forces massive localized instrumentation noise** (predominantly the severe electrical fault noise driving `Sensor_4`) straight into the supposedly clean structural monitoring subspace[cite: 260, 261, 282]. [cite_start]This severely degrades the signal-to-noise ratio of the diagnostics layer, compromising its baseline integrity[cite: 282].

---

### Question 4: Operational Trade-offs in System Health Monitoring

#### 1. Core Comparison of Monitoring Strategies
* [cite_start]**The PCA Strategy ($T^2$ + $Q$):** Monitors asset integrity using two separate bounding layers[cite: 284]. [cite_start]Hotelling's $T^2$ tracks normal operational shifts within the dominant 2D subspace, while the $Q$ statistic tracks out-of-subspace residual errors to detect changes in the underlying correlation matrix[cite: 100, 284].
* [cite_start]**The FA Strategy (Direct Latent Scores):** Monitors the rotated, decoupled physical process dimensions ($f_1$ and $f_2$) directly in the latent space by checking for drifts or anomalies directly on the factor score states[cite: 255, 285].

#### 2. Strategy Selection for Sensor Fault Robustness
[cite_start]The **Factor Analysis (FA) Strategy** is significantly more robust against a single sensor losing calibration or experiencing an electrical short circuit[cite: 286].

#### 3. Technical Justification using Sensor Uniqueness ($\varphi^2$) Metrics
[cite_start]In an operational environment, if a single instrument like `Sensor_4` experiences a massive electrical short or calibration failure, its localized variance spikes severely[cite: 260, 261, 268].

* [cite_start]In the **PCA Strategy**, this massive localized energy spike instantly distorts the system[cite: 273]. [cite_start]Because PCA cannot distinguish types of variance, the localized fault will bleed directly into the principal eigenvector calculations, inflating the eigenvalues, shifting the principal axes, and throwing both the $T^2$ and $Q$ monitors completely out of bounds—resulting in an immediate false alarm for the entire plant[cite: 100, 273, 284].
* [cite_start]In the **FA Strategy**, the generative model explicitly maps a dedicated error parameter for each channel, known as the **Sensor Uniqueness Noise Floor ($\varphi^2$)**[cite: 124, 230, 245]. [cite_start]When `Sensor_4` breaks, the FA engine isolates this localized white noise spike, pushing its Uniqueness value ($\varphi^2$) close to $100\%$ ($>98\%$)[cite: 261, 271]. [cite_start]Because the covariance modeling equations partition this uniqueness matrix ($\Psi$) diagonally away from the shared subspace, the underlying structural loading calculations ($\Lambda \Lambda^T$) remain completely unaffected[cite: 128, 142].

[cite_start]Thus, the plant operator can monitor the true physical states ($f_1$ and $f_2$) with perfect stability on the dashboard, completely undisturbed by the broken sensor hardware[cite: 256, 285, 286].